src/MongoDB/Command.c

```diff
@@ -194,7 +194,11 @@ static HashTable* php_phongo_command_get_debug_info(zval* object, int* is_temp T
 		zval* zv;
 #endif
 
-		php_phongo_bson_to_zval(bson_get_data(intern->bson), intern->bson->len, &zv);
+		if (!php_phongo_bson_to_zval(bson_get_data(intern->bson), intern->bson->len, &zv)) {
+			zval_ptr_dtor(&zv);
+			return Z_ARRVAL(retval);
```

Comment:
Thoughts on using `goto done;` and adding a `done:` label above the original `return Z_ARRVAL(retval);` line at the end of this function? It'd be closer to what we do in other places with failure/cleanup jumps.

I wouldn't suggest this for cases where we're simply returning (e.g. `Javascript::getScope()`), but I think helps avoid duplication when we're returning an expression, which could get out of sync. If so, other `get_debug_info` handlers could get the same treatment.

Happy to defer to you on this. You're also allowed to berate me for publicly advocating for a `goto` statement :)

Reply:
![XKCD goto](https://user-images.githubusercontent.com/383198/67657321-b3c44980-f956-11e9-840d-0cd93bd95be5.png)
([Source: XKCD](https://xkcd.com/292))

That said, use of `goto` as a replacement for `finally` and failure returns is established in the driver, so I don't mind adding those here as well 👍